geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/membership/GMSJoinLeave.java

```diff
@@ -1152,7 +1152,14 @@ boolean findCoordinator() {
             }
           }
         } catch (IOException | ClassNotFoundException problem) {
-          logger.debug("EOFException IOException ", problem);
+          logger.debug("Exception thrown when contacting a locator", problem);
+          if (state.locatorsContacted == 0 && System.currentTimeMillis() < giveUpTime) {
+            try {
+              Thread.sleep(1000);
+            } catch (InterruptedException e) {
+              throw new SystemConnectException("Interrupted while trying to contact locators");
```

Comment:
It's a change from the previous behaviour, but I think we should set `Thread.interrupted` here, or at least put `e` as the cause of the exception thrown. Generally when a thread gets interrupted it's because a server is shutting down.